cmd/clusterawsadm/cmd/resource/list/list.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package list
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/session"
+	tagapi "github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/spf13/cobra"
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/api/v1alpha4"
+	"sigs.k8s.io/cluster-api-provider-aws/cmd/clusterawsadm/cmd/flags"
+	cmdout "sigs.k8s.io/cluster-api-provider-aws/cmd/clusterawsadm/printers"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/cmd"
+)
+
+func ListAWSResourceCmd() *cobra.Command {
+	outputPrinter := ""
+	clusterName := ""
+	region := ""
+	newCmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all AWS resources created by CAPA",
+		Long: cmd.LongDesc(`
+			List AWS resources created by CAPA based on region and cluster-name. If region and cluster-name
+			is not provided, then it will throw an error.
+		`),
+		Example: cmd.Examples(`
+		# List AWS resources created by CAPA in given region and clustername
+		clusterawsadm resource list --region=us-east-1 --cluster-name=test-cluster
+		`),
+		Args: cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			var err error
+			region, err = flags.GetRegion(cmd)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("Attempting to fetch resources created by CAPA for cluster:%s present in %s\n\n", clusterName, region)
+			return listAWSResource(&region, &clusterName, &outputPrinter)
+		},
+	}
+
+	newCmd.Flags().StringVarP(&region, "region", "r", "", "The AWS region where resources are created by CAPA")
+	newCmd.Flags().StringVarP(&clusterName, "cluster-name", "n", "", "The name of the cluster where AWS resources created by CAPA")
+	newCmd.Flags().StringVarP(&outputPrinter, "output", "o", "table", "The output format of the results. Possible values: table,json,yaml")
+	newCmd.MarkFlagRequired("cluster-name") //nolint: errcheck
+	return newCmd
+}
+
+func listAWSResource(region, clusterName, printerType *string) error {
```

Comment:
Move this function to cmd/clusterawsadm/cmd to something like cmd/clusterawsadm/resources and export the function. contents of cmd/clusterawsadm/cmd package should be mostly cobra related.